misc/ls-extensions/modules/performance-analyzer-services/src/main/java/io/ballerina/PerformanceAnalyzerNodeVisitor.java

```diff
@@ -307,6 +314,42 @@ public void visit(AssignmentStatementNode assignmentStatementNode) {
         assignmentStatementNode.expression().accept(this);
     }
 
+    @Override
+    public void visit(NamedWorkerDeclarationNode namedWorkerDeclarationNode) {
+
+        if (withinRange) {
+            withinWorker = true;
+            if (!isWorkerExists) {
+                isWorkerExists = namedWorkerDeclarationNode != null;
+            }
+            if (namedWorkerDeclarationNode != null) {
+                namedWorkerDeclarationNode.workerBody().accept(this);
+            }
+            withinWorker = false;
+        }
+    }
+
+    @Override
+    public void visit(WaitActionNode waitActionNode) {
+
+        waitActionNode.waitFutureExpr().accept(this);
+    }
+
+    @Override
+    public void visit(SimpleNameReferenceNode simpleNameReferenceNode) {
+
+        if (withinRange) {
+            Optional<Symbol> waitNode = model.symbol(simpleNameReferenceNode);
+            if (waitNode.isEmpty() || waitNode.get().getLocation().isEmpty()) {
+                return;
+            }
+            SymbolKind kind = waitNode.get().kind();
```

Comment:
Shall we move the kind inside the if block? We dont need the kind variable assignment if `!isWorkerWaiting`.

Reply:
Fixed